loadbalancer/algorithm.go

```diff
@@ -197,9 +201,66 @@ func (ch *consistentHash) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 	return ctx.Route.LBEndpoints[choice]
 }
 
+type powerOfRandomNChoices struct {
+	rand            *rand.Rand
+	numberOfChoices int
+}
+
+// getScore returns negative value of inflightrequests count.
+func (p *powerOfRandomNChoices) getScore(e routing.LBEndpoint) int {
+	// endpoints with higher inflight request should have lower score
+	return -e.Metrics.GetInflightRequests()
+}
+
+// getRandomIndex returns random integer with the range of 0 and l
+// and skips excluded integers. If the length is equal to excluded length
+// it behaves as if excluded list is empty
```

Comment:
Please have a look at https://github.com/zalando/skipper/pull/1507#discussion_r472041811 regarding exclusions.

Please consider borrowing [core algorithm](https://github.com/zalando/skipper/pull/1507/files#diff-8ef30ff734de1bf682a3b0d530f9f663R130-R147) from the previous attempt.

It might also be interesting [how it is implemented in haproxy](https://github.com/haproxy/haproxy/commit/21c741a665f)